src/main/scala/it/nerdammer/spark/hbase/HBaseWriterBuilder.scala

```diff
@@ -84,9 +86,27 @@ class HBaseWriter[R: ClassTag](builder: HBaseWriterBuilder[R])(implicit mapper:
         if(saltingProvider.isEmpty) rawRowKey
         else Bytes.toBytes(saltingProvider.get.salt(rawRowKey) + Bytes.toString(rawRowKey))
 
-      val put = new Put(rowKey)
+      var record = columnsNames.zip(columns)
+
+      val put = if (builder.timestampColumn != None) {
+        val findTimestampColumn = (item:(String, Option[Array[Byte]])) => item._1 ==  builder.timestampColumn.get
+
+        if (record.find(findTimestampColumn).isEmpty) {
+          throw new IllegalArgumentException(s"Timestamp column doesn't exists - ${ builder.timestampColumn.get} found ${columns.size}")
+        }
+
+        val timestamp = Bytes.toString(record.find(findTimestampColumn).get._2.get).toLong
+
+        record = record.dropWhile(findTimestampColumn)
```

Comment:
I'm not convinced by this. I think it does nothing, as the "timestamp" column is always the last one, nothing is removed. So I think the "timestamp" column will be written.

Instead of removing data from the record, why don't skip writing the timestamp column in the foreach block below. This way, users will be allowed to put the timestamp column everywhere, not just at the end of the list.

Reply:
I agree that I can include this skip in the loop below looks much better.

I changed the behavior to became possible to pass the "timestamp" column in any position with any name, actually is the user who defined the column name. 

```scala
 rdd.toHBaseTable("mytable")
     .toColumns("column1", "column2", "my_timestamp", "created_at")
     .inColumnFamily("mycf")
     .usingTimestampRecord("my_timestamp")
     .save()
``` 

What do you think?

Reply:
I also can create a test to make sure that when the timestamp record is defined will not be written to a column.